### PR TITLE
Move k8s script to docker-config-engine

### PR DESCRIPTION
**Pull request by lixiaoyuner**
#### Why I did it
To reduce the container's dependency from host system
##### Work item tracking
- Microsoft ADO **(number only)**:
17713469
#### How I did it
Move the k8s container startup script to config engine container, other than mount it from host.
#### How to verify it
Check file path(/usr/share/sonic/scripts/container_startup.py) inside config engine container.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by ganglyu — APPROVED** [review]
LGTM

**Review by ganglyu — APPROVED** [review]

File: dockers/docker-dhcp-relay/start.sh [hunk 1]
```diff
@@ -5,7 +5,7 @@ if [ "${RUNTIME_OWNER}" == "" ]; then
 fi
 
 CTR_SCRIPT="/usr/share/sonic/scripts/container_startup.py"
-if test -f ${CTR_SCRIPT}
+if [ $RUNTIME_OWNER = "kube" ] && test -f ${CTR_SCRIPT}
```
Comment (qiluo-msft) [inline dockers/docker-dhcp-relay/start.sh:8]: >test [](http://example.com/codeflow?start=34&length=4)

I am confused:
If $RUNTIME_OWNER=="kube" and CTR_SCRIPT does not exist, you run the script?

File: rules/config [hunk 2]
```diff
@@ -178,7 +178,7 @@ INCLUDE_ROUTER_ADVERTISER ?= y
 
 # INCLUDE_KUBERNETES - if set to y kubernetes packages are installed to be able to
 # run as worker node in kubernetes cluster.
-INCLUDE_KUBERNETES ?= n
+INCLUDE_KUBERNETES ?= y
```
Comment (qiluo-msft) [inline rules/config:181]: >y [](http://example.com/codeflow?start=22&length=1)

This is behavior change. Other changes in this file is refactoring. Could you separate into multiple PRs?

File: src/sonic-ctrmgrd/ctrmgr/container [hunk 3]
```diff
@@ -11,7 +11,7 @@ import datetime
 import docker
 from swsscommon import swsscommon
 
-CTR_STATE_SCR_PATH = '/usr/share/sonic/scripts/container_startup.py'
+CTR_STATE_SCR_PATH = '/etc/sonic/remote_ctr.config.json'
```
Comment (qiluo-msft) [inline src/sonic-ctrmgrd/ctrmgr/container:14]: >remote_ctr.config.json [](http://example.com/codeflow?start=33&length=22)

Original it is a script, and name SCR is script. Why it is now a json file?

File: dockers/docker-dhcp-relay/start.sh (hunk 1 above)
Comment (lixiaoyuner) [reply dockers/docker-dhcp-relay/start.sh:8]: This code means if $RUNTIME_OWNER=="kube" and CTR_SCRIPT exists, than I run the script

File: rules/config (hunk 2 above)
Comment (lixiaoyuner) [reply rules/config]: Yes, I just enable it for this moment to check PR build tests pass rate. Should disable it as default.

File: src/sonic-ctrmgrd/ctrmgr/container (hunk 3 above)
Comment (lixiaoyuner) [reply src/sonic-ctrmgrd/ctrmgr/container:14]: The logic here is to check whether k8s is enabled, if the file exists, it means enabled. So just check file whether exists, don't care the content.

**Review by ganglyu — APPROVED** [review]

File: files/build_templates/docker_image_ctl.j2 [hunk 4]
```diff
@@ -578,7 +578,6 @@ start() {
 {%- endif -%}
 {%- if docker_container_name == "bgp" %}
         -v /etc/sonic/frr/$DEV:/etc/frr:rw \
-        -v /usr/share/sonic/scripts:/usr/share/sonic/scripts:ro \
```
[the remark this answers is not in the document]
Comment (lixiaoyuner) [reply files/build_templates/docker_image_ctl.j2:581]: No other files under the path, it's just for k8s usage.

File: src/sonic-ctrmgrd/ctrmgr/container_startup.py [hunk 5]
```diff
@@ -216,6 +216,9 @@ def container_up(feature, owner, version):
     debug_msg("args: feature={}, owner={}, version={} DB: set_owner={} state_data={}".format(
         feature, owner, version, set_owner, json.dumps(state_data, indent=4)))
 
+    if state_data[SYSTEM_STATE] == '':
```
[the remark this answers is not in the document]
Comment (lixiaoyuner) [reply src/sonic-ctrmgrd/ctrmgr/container_startup.py:219]: After we move the k8s script inside container, it means we will run the script each time the container start up. But we don't need to run the script when k8s feature is not included. So, add this logic, when the SYSTEM_STATE is empty, it means the k8s feature is not included, the script will return directly and doing nothing.

**Review by losha228 — APPROVED** [review]

File: src/sonic-ctrmgrd/ctrmgr/container (hunk 3 above)
Comment (qiluo-msft) [reply src/sonic-ctrmgrd/ctrmgr/container]: Do you also want to change CTR_STATE_SCR_PATH -> CTRMGRD_SERVICE_PATH

Comment (lixiaoyuner) [reply src/sonic-ctrmgrd/ctrmgr/container]: Thanks, changed.

**Review by qiluo-msft — DISMISSED** [review]

**Review by lguohan — DISMISSED** [review]

**Comment by lguohan** [comment]
@lixiaoyuner , can you resolve conflict?

**Comment by lixiaoyuner** [comment]
> @lixiaoyuner , can you resolve conflict?

@lguohan , Thanks very much for your approval. I fixed the conflict.

File: files/build_templates/docker_image_ctl.j2 (hunk 4 above)
[the remark this answers is not in the document]
Comment (lguohan) [reply files/build_templates/docker_image_ctl.j2:582]: @lixiaoyuner , can you check this pr? 6a54bc439a980c7fc10e7a74de6472ee7731f8a5 which introduces this change, and it is not related to container manage

[the remark this answers is not in the document]
Comment (lixiaoyuner) [reply files/build_templates/docker_image_ctl.j2:582]: I logged in the latest bgp container to check, just only k8s script is there. Let me ask @zbud-msft to double confirm.

**Review by qiluo-msft — APPROVED** [review]

File: files/build_templates/docker_image_ctl.j2 (hunk 4 above)
[the remark this answers is not in the document]
Comment (zbud-msft) [reply files/build_templates/docker_image_ctl.j2:582]: Yes, we can remove that line. We no longer use scripts folder for rsyslog plugin as we have created it as a deb package instead. 

https://github.com/sonic-net/sonic-buildimage/pull/12059/files#diff-f65ade0922278f02fe94b7232f322776492d4bfbd2d43203118b834d97b06dfbR15

**Pull request merged**